lib/plug/conn.ex

```diff
@@ -934,6 +934,44 @@ defmodule Plug.Conn do
   end
 
   @doc """
+  Push a resource to the client.
+
+  If the adapter does not support server push then this is a noop.
+  """
+  @spec push(t, String.t, Keyword.t) :: t
+  def push(conn, path, headers \\ []) do
+    adapter_push(conn, path, headers)
+    conn
+  end
+
+  @doc """
+  Push a resource to the client.
+
+  If the adapter does not support server push then this function will raise.
+  """
+  @spec push!(t, String.t, Keyword.t) :: t
+  def push!(%Conn{adapter: {adapter, _}} = conn, path, headers \\ []) do
+    case adapter_push(conn, path, headers) do
+      :ok ->
+        conn
+
+      _ ->
+        raise "server push not supported by #{inspect(adapter)}." <>
+        "You should either delete the call to `push!/3` or switch to an " <>
+        "adapter that does support server push such as Plug.Adapters.Cowboy2."
+    end
+  end
+
+  defp adapter_push(%Conn{adapter: {adapter, payload}}, path, headers) do
+    headers =
+      case Enum.find(headers, &Kernel.match?({"accept", _}, &1)) do
```

Comment:
List.keyfind is quite faster.

Reply:
Thanks, I thought there may be a better way!